lib/manager/mix/artifacts.js

```diff
@@ -0,0 +1,112 @@
+const upath = require('upath');
+const fs = require('fs-extra');
+const { exec } = require('child-process-promise');
+
+module.exports = {
+  getArtifacts,
+};
+
+async function getArtifacts(
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config
+) {
+  await logger.debug(`mix.getArtifacts(${packageFileName})`);
+  if (updatedDeps === undefined || updatedDeps.length < 1) {
+    logger.debug('No updated mix deps - returning null');
+    return null;
+  }
+  if (!config.gitFs) {
+    logger.warn(
+      'Renovate administrator should enable gitFs in order to support mix.lock updating'
+    );
+    return null;
+  }
+  if (!config.localDir) {
+    logger.debug('No local dir specified');
+    return null;
+  }
+  const lockFileName = 'mix.lock';
+  try {
+    const localPackageFileName = upath.join(config.localDir, packageFileName);
+    await fs.outputFile(localPackageFileName, newPackageFileContent);
+  } catch (err) {
+    logger.warn({ err, message: err.message }, 'mix.exs is not found');
+    return {
+      lockFileError: {
+        lockFile: lockFileName,
+        stderr: err.message,
+      },
+    };
+  }
+
+  const existingLockFileContent = await platform.getFile(lockFileName);
```

Comment:
This will always fetch the `mix.lock` in the root of the repo. Is that your intention? What about projects or package files that are in subdirectories?

Reply:
it fings exact lockFile of main project or subproject to update only found deps

Reply:
So this solution does not support a repo that has a `mix.lock` anywhere except the root of the repository?

Reply:
e.g.: https://github.com/konstantinkurganov-test/umbrella
we have a root repo contaning `mix.exs` and `mix.lock`
this repo also has subprojects that also contain their own `mix.exs` and `mix.lock`
this solution supports processing of all these files

A mix project should be created via `mix new` and in this case `mix.exs` and `mix.lock` are created in the root of the project

Reply:
With `const lockFileName = 'mix.lock';` then `const existingLockFileContent = await platform.getFile(lockFileName);`. How would this work if the `mix.exs` is updated in a subdirectory?

Reply:
Sorry, I was wrong. I finally understood the point. Dealing with umbrella projects we have a `mix.exs` file in each subdirectory but we have have a _mix.lock_ only in _root_ directory, containing artifacts for all submodules. So there is no need to update lockfile in subdirs.